controllers/vcdmachine_controller.go

```diff
@@ -888,8 +888,10 @@ func (r *VCDMachineReconciler) reconcileVMNetworks(vdcManager *vcdsdk.VdcManager
 
 		err = vm.UpdateNetworkConnectionSection(connections)
 		if err != nil {
-			return errors.Errorf("failed to update networks of VM")
+			return errors.Wrapf(err, "failed to update networks of VM")
 		}
+		// update vm.VM object for the rest of the flow, especially for getPrimaryNetwork function
+		vm.VM.NetworkConnectionSection = connections
```

Comment:
I think refreshing the vm object using vm.Refresh() would be better.

Reply:
but it will send an additional request to the API. Is it desired?